File: exercises/js/day2-step1-setup-account.js
// ============================================================
// Day 2 - Step 1 Exercise: Setup Account (Enable Default Ripple)
// Learn: How to configure account flags on the XRP Ledger.
// Default Ripple is required before issuing tokens.
// Reference materials:
// - https://learn.xrpl-commons.org/course/code-with-the-xrpl/
// - https://xrpl.org/docs/references/protocol/transactions/types/accountset
// - https://js.xrpl.org/
// ============================================================

const xrpl = require('xrpl');
const fs = require('fs');
const { notifyProgress } = require('../../scripts/notify');

const TESTNET_URL = 'wss://s.altnet.rippletest.net:51233';

async function main() {
  const { wallets } = JSON.parse(fs.readFileSync('../../wallets.json', 'utf-8'));
  const issuerWallet = xrpl.Wallet.fromSeed(wallets[0].seed);

  const client = new xrpl.Client(TESTNET_URL);
  await client.connect();
  console.log('Connected to XRPL Testnet');

  // ╔══════════════════════════════════════════════════════════════════╗
  // ║ TODO 1: Submit an AccountSet transaction to enable Default       ║
  // ║         Ripple on the issuer account                             ║
  // ║                                                                  ║
  // ║ HINT: {                                                          ║
  // ║   TransactionType: 'AccountSet',                                 ║
  // ║   Account: issuerWallet.address,                                 ║
  // ║   SetFlag: xrpl.AccountSetAsfFlags.asfDefaultRipple              ║
  // ║ }                                                                ║
  // ║                                                                  ║
  // ║ Submit: client.submitAndWait(tx, { wallet: issuerWallet })       ║
  // ╚══════════════════════════════════════════════════════════════════╝
  // YOUR CODE HERE

  console.log('Default Ripple enabled on issuer account!');
  console.log(`\nExplorer: https://testnet.xrpl.org/accounts/${issuerWallet.address}`);
  await notifyProgress('Day 2 Step 1: Setup Account');
  await client.disconnect();
}

main().catch(console.error);

File: exercises/python/day2_step1_setup_account.py
"""
Day 2 Step 1 Exercise: Setup Account
=====================================
Enable Default Ripple flag on the issuer account.

Reference materials:
- https://learn.xrpl-commons.org/course/code-with-the-xrpl/
- https://xrpl.org/docs/references/protocol/transactions/types/accountset
- https://xrpl-py.readthedocs.io/en/stable/
"""

import json
from xrpl.clients import WebsocketClient
from xrpl.wallet import Wallet
from xrpl.models.transactions import AccountSet
from xrpl.transaction import submit_and_wait
import sys; sys.path.insert(0, '../../scripts')
from notify_progress import notify_progress

TESTNET_URL = 'wss://s.altnet.rippletest.net:51233'

# Load wallets
with open('../../wallets.json') as f:
    data = json.load(f)
    wallets = data['wallets']

issuer = Wallet.from_seed(wallets[0]['seed'])
print(f"Issuer: {issuer.address}")

with WebsocketClient(TESTNET_URL) as client:

    # ╔══════════════════════════════════════════════════════════════════╗
    # ║ TODO 1: Enable Default Ripple (flag 8) on the issuer account    ║
    # ║                                                                  ║
    # ║ HINT: AccountSet(account=issuer.address, set_flag=8)             ║
    # ╚══════════════════════════════════════════════════════════════════╝
    account_set_tx = None  # Replace with AccountSet(...)
    # result = submit_and_wait(account_set_tx, client, issuer)
    # print(result)

# Explorer: https://testnet.xrpl.org/accounts/{address}
print(f"\nExplorer: https://testnet.xrpl.org/accounts/{issuer.address}")
notify_progress('Day 2 Step 1: Setup Account')
